### ISSUE TITLE
[Bug] MoltenExportOutput "attempt to index local 'opts' (a number value)"

### ISSUE THREAD

**Issue by EmDash00** [benlubas/molten-nvim#181]
- OS: Ubuntu 22.04.4 LTS
- NeoVim Version: v0.10.0-dev-2786+gb3f9da952
- Python Version: 3.10
- Python is installed with: apt
- Health checks

<details>
  <summary>`:checkhealth molten`</summary>

```
molten: require("molten.health").check()

molten-nvim ~
- OK NeoVim >=0.9
- OK Python >=3.10
- OK Python module pynvim found
- OK Python module jupyter-client found
- OK Python module cairosvg found
- OK Python module pnglatex found
- OK Python module plotly found
- OK Python module kaleido found
- OK Python module pyperclip found
- OK Python module nbformat found
- OK Python module pillow found
```

</details>

<details>
  <summary>`:checkhealth provider` (the python parts)</summary>

```
provider.python: require("provider.python.health").check()

Python 3 provider (optional) ~
- pyenv: Path: /home/ember/.pyenv/libexec/pyenv
- pyenv: Root: /home/ember/.pyenv
- Using: g:python3_host_prog = "/home/ember/.config/virtualenvs/neovim/bin/python3"
- Executable: /home/ember/.config/virtualenvs/neovim/bin/python3
- Python version: 3.10.12
- pynvim version: 0.5.0
- OK Latest pynvim is installed.

Python virtualenv ~
- OK no $VIRTUAL_ENV
```

</details>

## Description

Using the configuration given for automatically saving outputs to a jupyter notebook, I get the following massive error string.


```
Error detected while processing BufWriteCmd Autocommands for "<buffer=1>"..BufWritePost Autocommands for "*.ipynb":                                                                                                                                             
Error executing lua callback: vim/_editor.lua:0: BufWriteCmd Autocommands for "<buffer=1>"..BufWritePost Autocommands for "*.ipynb"..script nvim_exec2() called at BufWritePost Autocommands for "*.ipynb":0..function remote#define#CommandBootstrap[5]..remote
#define#request, line 2: Vim(let):Error invoking '/home/ember/.local/share/nvim/site/pack/rocks/start/molten-nvim/rplugin/python3/molten:command:MoltenExportOutput' on channel 6 (python3-rplugin-host):                                                       
error caught in request handler '/home/ember/.local/share/nvim/site/pack/rocks/start/molten-nvim/rplugin/python3/molten:command:MoltenExportOutput [[], 1]':                                                                                                    
Traceback (most recent call last):                                                                                                                                                                                                                              
  File "/home/ember/.local/share/nvim/site/pack/rocks/start/molten-nvim/rplugin/python3/molten/utils.py", line 11, in inner                                                                                                                                     
    func(self, *args, **kwargs)                                                                                                                                                                                                                                 
  File "/home/ember/.local/share/nvim/site/pack/rocks/start/molten-nvim/rplugin/python3/molten/__init__.py", line 754, in command_export                                                                                                                        
    self.kernel_check(f"MoltenExportOutput{'!' if bang else ''} {path} %k", buf)                                                                                                                                                                                
  File "/home/ember/.local/share/nvim/site/pack/rocks/start/molten-nvim/rplugin/python3/molten/__init__.py", line 599, in kernel_check                                                                                                                          
    self.nvim.command(c)                                                                                                                                                                                                                                        
  File "/home/ember/.config/virtualenvs/neovim/lib/python3.10/site-packages/pynvim/api/nvim.py", line 311, in command                                                                                                                                           
    return self.request('nvim_command', string, **kwargs)                                                                                                                                                                                                       
  File "/home/ember/.config/virtualenvs/neovim/lib/python3.10/site-packages/pynvim/api/nvim.py", line 199, in request                                                                                                                                           
    res = self._session.request(name, *args, **kwargs)                                                                                                                                                                                                          
  File "/home/ember/.config/virtualenvs/neovim/lib/python3.10/site-packages/pynvim/msgpack_rpc/session.py", line 139, in request                                                                                                                                
    raise self.error_wrapper(err)                                                                                                                                                                                                                               
pynvim.api.common.NvimError: BufWriteCmd Autocommands for "<buffer=1>"..BufWritePost Autocommands for "*.ipynb"..script nvim_exec2() called at BufWritePost Autocommands for "*.ipynb":0..function remote#define#CommandBootstrap[5]..remote#define#request[2]..
remote#define#request, line 2: Vim(let):Error invoking '/home/ember/.local/share/nvim/site/pack/rocks/start/molten-nvim/rplugin/python3/molten:command:MoltenExportOutput' on channel 6 (python3-rplugin-host):                                                 
error caught in request handler '/home/ember/.local/share/nvim/site/pack/rocks/start/molten-nvim/rplugin/python3/molten:command:MoltenExportOutput [['/home/ember/Documents/Work/test/test.ipynb', 'test'], 1]':                                                
Traceback (most recent call last):                                                                                                                                                                                                                              
  File "/home/ember/.local/share/nvim/site/pack/rocks/start/molten-nvim/rplugin/python3/molten/utils.py", line 11, in inner                                                                                                                                     
    func(self, *args, **kwargs)                                                                                                                                                                                                                                 
  File "/home/ember/.local/share/nvim/site/pack/rocks/start/molten-nvim/rplugin/python3/molten/__init__.py", line 759, in command_export                                                                                                                        
    export_outputs(self.nvim, molten, path, bang)                                                                                                                                                                                                               
  File "/home/ember/.local/share/nvim/site/pack/rocks/start/molten-nvim/rplugin/python3/molten/ipynb.py", line 172, in export_outputs                                                                                                                           
    if compare_contents(nvim, nb_cell, code_cell, lang):                                                                                                                                                                                                        
  File "/home/ember/.local/share/nvim/site/pack/rocks/start/molten-nvim/rplugin/python3/molten/ipynb.py", line 213, in compare_contents                                                                                                                         
    clean_nb = nvim.lua._remove_comments(nb_cell["source"] + "\n", lang)                                                                                                                                                                                        
  File "/home/ember/.config/virtualenvs/neovim/lib/python3.10/site-packages/pynvim/api/nvim.py", line 599, in __call__                                                                                                                                          
    return self._nvim.exec_lua(code, *args, **kwargs)                                                                                                                                                                                                           
  File "/home/ember/.config/virtualenvs/neovim/lib/python3.10/site-packages/pynvim/api/nvim.py", line 344, in exec_lua                                                                                                                                          
    return self.request('nvim_execute_lua', code, args, **kwargs)                                                                                                                                                                                               
  File "/home/ember/.config/virtualenvs/neovim/lib/python3.10/site-packages/pynvim/api/nvim.py", line 199, in request                                                                                                                                           
    res = self._session.request(name, *args, **kwargs)                                                                                                                                                                                                          
  File "/home/ember/.config/virtualenvs/neovim/lib/python3.10/site-packages/pynvim/msgpack_rpc/session.py", line 139, in request                                                                                                                                
    raise self.error_wrapper(err)                                                                                                                                                                                                                               
pynvim.api.common.NvimError: Error executing lua: ...-linux64/share/nvim/runtime/lua/vim/treesitter/query.lua:935: attempt to index local 'opts' (a number value)                                                                                               
stack traceback:                                                                                                                                                                                                                                                
        ...-linux64/share/nvim/runtime/lua/vim/treesitter/query.lua:935: in function 'iter_matches'                                                                                                                                                             
        ...ite/pack/rocks/start/molten-nvim/lua/remove_comments.lua:17: in function <...ite/pack/rocks/start/molten-nvim/lua/remove_comments.lua:8>                                                                                                             
        [C]: in function 'nvim_exec2'                                                                                                                                                                                                                           
        vim/_editor.lua: in function 'cmd'                                                                                                                                                                                                                      
        /home/ember/dotfiles/.config/nvim/plugin/molten-nvim.lua:109: in function </home/ember/dotfiles/.config/nvim/plugin/molten-nvim.lua:107>                                                                                                                
        [C]: in function 'nvim_exec_autocmds'                                                                                                                                                                                                                   
        ...ite/pack/rocks/start/jupytext.nvim/lua/jupytext/init.lua:31: in function 'write_to_ipynb'                                                                                                                                                            
        ...ite/pack/rocks/start/jupytext.nvim/lua/jupytext/init.lua:118: in function <...ite/pack/rocks/start/jupytext.nvim/lua/jupytext/init.lua:117>                                                                                                          
stack traceback:                                                                                                                                                                                                                                                
        [C]: in function 'nvim_exec2'                                                                                                                                                                                                                           
        vim/_editor.lua: in function 'cmd'                                                                                                                                                                                                                      
        /home/ember/dotfiles/.config/nvim/plugin/molten-nvim.lua:109: in function </home/ember/dotfiles/.config/nvim/plugin/molten-nvim.lua:107>                                                                                                                
        [C]: in function 'nvim_exec_autocmds'                                                                                                                                                                                                                   
        ...ite/pack/rocks/start/jupytext.nvim/lua/jupytext/init.lua:31: in function 'write_to_ipynb'                                                                                                                                                            
        ...ite/pack/rocks/start/jupytext.nvim/lua/jupytext/init.lua:118: in function <...ite/pack/rocks/start/jupytext.nvim/lua/jupytext/init.lua:117> 
```

The relevant part is probably

```
pynvim.api.common.NvimError: Error executing lua: ...-linux64/share/nvim/runtime/lua/vim/treesitter/query.lua:935: attempt to index local 'opts' (a number value)
```

## Reproduction Steps
I'm using the new [rocks.nvim](https://github.com/nvim-neorocks/rocks.nvim) package manager. Install that first.

Pip Freeze Output:
```
attrs==23.2.0
beautifulsoup4==4.12.3
bleach==6.1.0
cairocffi==1.6.1
CairoSVG==2.7.1
cffi==1.16.0
cssselect2==0.7.0
defusedxml==0.7.1
fastjsonschema==2.19.1
greenlet==3.0.3
Jinja2==3.1.3
jsonschema==4.21.1
jsonschema-specifications==2023.12.1
jupyter_client==8.6.0
jupyter_core==5.7.1
jupyterlab_pygments==0.3.0
jupytext==1.16.1
kaleido==0.2.1
markdown-it-py==3.0.0
MarkupSafe==2.1.5
mdit-py-plugins==0.4.0
mdurl==0.1.2
mistune==3.0.2
msgpack==1.0.7
nbclient==0.10.0
nbconvert==7.16.3
nbformat==5.10.4
packaging==23.2
pandocfilters==1.5.1
pillow==10.2.0
platformdirs==4.2.0
plotly==5.19.0
pnglatex==1.1
pycparser==2.21
Pygments==2.17.2
pynvim==0.5.0
pyperclip==1.8.2
python-dateutil==2.8.2
PyYAML==6.0.1
pyzmq==25.1.2
referencing==0.34.0
rpds-py==0.18.0
six==1.16.0
soupsieve==2.5
tenacity==8.2.3
tinycss2==1.2.1
toml==0.10.2
tornado==6.4
traitlets==5.14.1
webencodings==0.5.1
```

I put a minimal...ish config on [EmDash00/molten_minimal](https://github.com/EmDash00/molten_minimal). Once Rocks is installed, you can just copy that in and then run:

```
:Rocks sync
:UpdateRemotePlugins
```

(Unfortunately Rocks doesn't yet have post-build proccesses... though it's supposed to be handled by the plugin which this one isn't yet published to Luarocks :shrug:.

## Expected Behavior
Not a giant error message?

**Comment by benlubas** [comment]
Yeah that functionality hasn't worked on nightly I think ever. I ran stable back when I wrote it. I'll make sure this is working before the end of the month (when v0.10.0 is set to release), but I'm too busy to do it this week. If you want to take a look, this is the line causing the error https://github.com/benlubas/molten-nvim/blob/66b11de7c3132dcc4521d50039ce2b5f81e64ca2/lua/remove_comments.lua#L17
